### DISCUSSION TITLE
Design Decisions for Anisha

### DISCUSSION THREAD

**Discussion by DhanyaHerath** [msupply-foundation/open-msupply#1985] [General]
Trying this out as a way to store design decisions which should go past Anisha!

**Comment by DhanyaHerath** [comment]
#1912 [Need a good way to alert users about why they can't use certain functionality (eg. bc of permissions) #1912](https://github.com/openmsupply/open-msupply/issues/1912)

**Reply by andreievg** [reply]
@DhanyaHerath @anisha-msupply @richardmoizeau @mark-prins 

I was wondering, was this formalised somewhere, do we have general design guidelines/patterns documentation for omSupply ? 

This is slightly related to the next comment

**Comment by andreievg** [comment]
#### Andrei:
@anisha-msupply @richardmoizeau

Hi I wonder what our position is regarding hover over tooltips ? Are they accessible somehow in mobile ? 
And since most of our sites would eventually be mobile sites, and I think we are targeting mobile users in UX/UI decision, was wondering what is the alternative/strategy on that front ?

#### Jessica Jago:
Feedback from me: tablet users found it really hard.

#### Jessica Jago:
couldn't get the finger touch quite right to use them

#### Clemens:
Same here for sync errors (plus the tooltip disappeared before I was able to read it fully)

#### Mark Prins:
you can long-press on tooltippable elements to see the tooltips. there's no way of knowing that's a thing though

the trouble with tooltips is also mentioned in issues https://github.com/msupply-foundation/open-msupply/issues/1912 https://github.com/msupply-foundation/open-msupply/issues/2945

the 2945 issue is the 'do something better with login errors' one, which should also address the tooltip info icon, which is the one most people complain about.

#### Mark Prins:
we have tooltip info icon in outbound shipment line allocation modal too, but I don't think anyone has used them there. 

the other place is the sync page, and there's space on that one to write out a better error

#### James Brunskill:

For the why is my button disabled problem, I wonder if we could create a component that looks visually disabled, but on click shows a tooltip/callout?

![2024-02-22 14 54 52](https://github.com/msupply-foundation/open-msupply/assets/26194949/bddff6cf-c1ca-47d0-a08f-31a62654facd)

Probably needs a close icon option…

#### Andrei:
> you can long-press on tooltippable elements to see the tooltips

There has been various comments that this is too hard to do (from our support team, from Jess etc..), I am not sure if that's the UX we are aiming for ?

> there's no way of knowing that's a thing though

That's kind of another point about tooltips (I wanted to mention it here since i saw a tooltip being added to drop down element)

There are also tooltips on columns (and column click is both sort and tooltip ?)

I'll make a discussion out of this thread, and see what @RiFeryn and @arpanchia can suggest

#### Mark Prins:
toast? physically disconnected tho. if not acceptable, something similar with a disappearing popover. and the button could shake to indicate 'not allowed'

https://mui.com/material-ui/react-popover/ maybe?

Actually looks like tooltip can still be used https://mui.com/material-ui/react-tooltip/#triggers

#### Adam Dewey:
Maybe an icon to indicate you can long press something?

#### Adam Dewey:
And make the pressable area larger

**Comment by GaryWilletts** [comment]
Apparently long pressing on tooltipped elements shows the tooltip. Although nothing indicates this in the UI, it's just a training thing. People will easily remember "if I need more info, I just long press it and see if some more info pops up".

Another thought: we have little "i" icons on headers that show a tooltip on press. They would be good to include everywhere for tooltips. Might clutter the interface but we could style it to be less obtrusive, just visible so you know where you can press. Also might be able to include it inside a button?

Personally I like long press. No need to show anything on the UI, people just know (via training) that they can long press things to see more info. Lots of apps I use do this for info or functionality (e.g. Telegram) and it becomes second nature quite quickly. I suspect that users who don't get/like it are just saying they don't like change.
